src/fs.js

```diff
@@ -1,24 +1,16 @@
 'use strict';
 
-var File = require('./file');
 var fs = require('fs');
 var path = require('path');
 
-function Fs ($matcher, $transformer) {
-  this._cache = {};
+function Fs () {
   this._map = {};
-  this._matcher = $matcher;
-  this._transformer = $transformer;
 }
 
 Fs.prototype = {
-  cached: function (file) {
-    return this._cache[this.resolve(file)];
-  },
-
-  file: function (file) {
-    var resolved = this.resolve(file);
-    return this._cache[resolved] || (this._cache[resolved] = new File(this, this._matcher, this._transformer, resolved));
+  ext: function (file, ext, exts) {
+    exts = exts || [ext];
+    return exts.indexOf((path.extname(file) || '.').split('.')[1]) === -1 ? file + '.' + ext : file;
```

Comment:
What's the intention of `ext()`?

By my reading if it matches and of the `exts` it will always give you back the `ext` you specified rather than the one that it matched — by design?